tests/python/pants_test/engine/test_isolated_process.py

```diff
@@ -104,37 +126,52 @@ def test_blows_up_on_unhashable_args(self):
 
 
 class IsolatedProcessTest(SchedulerTestBase, unittest.TestCase):
-
   def test_integration_concat_with_snapshot_subjects_test(self):
     scheduler = self.mk_scheduler_in_example_fs([
       # subject to files / product of subject to files for snapshot.
       SnapshottedProcess.create(product_type=Concatted,
-                                binary_type=ShellCatToOutFile,
-                                input_selectors=(Select(Snapshot),),
-                                input_conversion=file_list_to_args_for_cat_with_snapshot_subjects_and_output_file,
-                                output_conversion=process_result_to_concatted_from_outfile),
+        binary_type=ShellCatToOutFile,
+        input_selectors=(Select(Snapshot),),
+        input_conversion=file_list_to_args_for_cat_with_snapshot_subjects_and_output_file,
+        output_conversion=process_result_to_concatted_from_outfile),
       SingletonRule(ShellCatToOutFile, ShellCatToOutFile()),
     ])
 
     request = scheduler.execution_request([Concatted],
-                                          [PathGlobs.create('', include=['fs_test/a/b/*'])])
-
+      [PathGlobs.create('', include=['fs_test/a/b/*'])])
     root_entries = scheduler.execute(request).root_products
     self.assertEquals(1, len(root_entries))
     state = self.assertFirstEntryIsReturn(root_entries, scheduler)
     concatted = state.value
 
     self.assertEqual(Concatted('one\ntwo\n'), concatted)
 
+  def test_javac_version_example(self):
+    sources = PathGlobs.create('', include=['inputs/src/java/simple/Simple.java'])
```

Comment:
Possibly not the right place to be asking but... Why do we need to use a glob here? Is it because it's the only way to turn a file into a `Stat` when handing it to the engine? It seems kind of a weird UX to be globbing to point at a single file (particularly as glob has the property where if the file is missing, that's silently ignored)...

Reply:
Yes, currently the only way. A bit of sugar to create an "infallible" PathGlobs instance that handles erroring for a miss would work.